sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,92 +16,286 @@
+      alluxioCmd = conf.getAlluxioCmd.getOrElse(
+        Seq("su", "ubuntu", "-c", s"$alluxioHome/bin/alluxio"))
+
+      if (!isInit) {
+        // Default to read from /opt/alluxio-2.8.0 if not setting ALLUXIO_HOME
+        var alluxio_port: String = null
+        var alluxio_master: String = null
+        var buffered_source: BufferedSource = null
+        try {
+          buffered_source = Source.fromFile(alluxioHome + "/conf/alluxio-site.properties")
+          val prop : Properties = new Properties()
+          prop.load(buffered_source.bufferedReader())
+          alluxio_master = prop.getProperty("alluxio.master.hostname")
+          alluxio_port = prop.getProperty("alluxio.master.rpc.port", "19998")
+        } catch {
+          case e: FileNotFoundException =>
+            throw new RuntimeException(s"Not found Alluxio config in " +
+              s"$alluxioHome/conf/alluxio-site.properties, " +
+              "please check if ALLUXIO_HOME is set correctly")
+        } finally {
+          if (buffered_source != null) buffered_source.close
+        }
+
+        if (alluxio_master == null) {
+          throw new RuntimeException(
+            s"Can't find alluxio.master.hostname from $alluxioHome/conf/alluxio-site.properties.")
+        }
+        alluxioMasterHost = Some(alluxio_master + ":" + alluxio_port)
+        // load mounted point by call Alluxio mount command.
+        val (ret, output) = runAlluxioCmd(" fs mount")
+        if (ret == 0) {
+          // parse the output, E.g.
+          // s3a://bucket-foo/        on  /bucket-foo
+          // s3a://bucket-another/    on  /bucket-another
+          // /local_path              on  /
+          for (line <- output) {
+            val items = line.trim.split(" +")
+            logDebug(line)
+            if (items.length >= 3) {
+              // if the first item contains the "://", it means it's a remote path.
+              // record it as a mounted point
+              if (items(0).contains("://")) {
+                mountedBuckets(items(2)) = items(0)
+                logInfo(s"Found mounted bucket ${items(0)} to ${items(2)}")
+              }
+            }
+          }
+        } else {
+          logWarning(s"Failed to run alluxio fs mount $ret")
+        }
+        isInit = true
+      }
+    }
+  }
+
+  // The path should be like s3://bucket/... or s3a://bucket/...
+  private def getSchemeAndBucketFromPath(path: String) : (String, String) = {
+    val i = path.split("://")
+    val scheme = i(0)
+    if (i.length <= 1) {
+      throw new RuntimeException(s"path $path is not expected for Alluxio auto mount")
+    }
+    val bucket = i(1).split("/")(0)
+    (scheme, bucket)
+  }
+
+  private def runAlluxioCmd(param : String) : (Int,
+    scala.collection.mutable.ArrayBuffer[String]) = {
+    val params = alluxioCmd.tails.collect {
+        case Seq(first, _, _*) => first
+        case Seq(last) => last + param
+      }.toSeq
+    logInfo(s"Run command ${params.last}")
```

Comment:
I don't htink this is guaranteed to be the command.  I want to make sure we aren't printing the secret key out here by accident so I would prefer to see this removed.